app/src/main/java/com/stevesoltys/seedvault/plugins/webdav/WebDavStorage.kt

```diff
@@ -169,6 +169,18 @@ internal abstract class WebDavStorage(
         }
     }
 
+    protected fun HttpException.isUnsupportedPropfind(): Boolean {
+        // nginx returns 400 for depth=2
+        if (code == 400) {
+            return true
+        }
+        // lighttpd returns 403 with <DAV:propfind-finite-depth/> error as if we used infinity
```

Comment:
FYI: lighttpd does support `Depth: infinity` but the feature is disabled by default since it can be slow on very large directory trees on very slow disks.

Since lighttpd 1.4.56 (released Nov 2020), you can add to lighttpd.conf:
`webdav.opts += ("propfind-depth-infinity" => "enable")`

Reply:
Nice to know, but does it also support off-spec stuff like `Depth: 2` like SabreDAV?

@t-m-w may be worth re-testing lighttpd with that option enabled.

Reply:
> but does it also support off-spec stuff like Depth: 2 like SabreDAV?

No, lighttpd mod_webdav does not support "off-spec" that I have not seen before.  Do you have a reference to SabreDAV doc describing the behavior?  (I can guess, but I'd rather not.)

From a (very, very, very quick) `grep` of https://github.com/sabre-io/dav, the code comments about `Depth: 2` for use with CalDAV.  The CalDAV spec is not currently implemented by lighttpd mod_webdav.

However, you can use lighttpd mod_webdav to serve file-backed collections, and you can have lighttpd route CalDAV url-paths back to nextcloud or owncloud sabredav-based backends.